src/machine/machine_atsamd51.go

```diff
@@ -2299,3 +2299,81 @@ func checkFlashError() error {
 
 	return nil
 }
+
+// Watchdog provides access to the hardware watchdog available
+// in the SAMD51.
+var Watchdog = &watchdogImpl{}
+
+const (
+	// WatchdogMaxSources is 8
+	//
+	// SAMD51 only supports a single source in hardware, so
+	// the support for multiple sources is in software.
+	WatchdogMaxSources = 8
+
+	// WatchdogMaxTimeout is 16s
+	WatchdogMaxTimeout = 16 // CYC16384/1024kHz
```

Comment:
Should these units be consistent with the configuration struct units? (Prefer smaller units, milliseconds)